pkg/exporter/prometheus_exporter.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package exporter
+
+import (
+	"net"
+	"net/http"
+	"strconv"
+
+	"contrib.go.opencensus.io/exporter/prometheus"
+	"github.com/golang/glog"
+	"go.opencensus.io/stats/view"
+
+	"k8s.io/node-problem-detector/cmd/options"
+	"k8s.io/node-problem-detector/pkg/types"
+)
+
+type prometheusExporter struct{}
+
+// NewPrometheusExporterOrDie creates an exporter to export metrics to Prometheus, panics if error occurs.
+func NewPrometheusExporterOrDie(npdo *options.NodeProblemDetectorOptions) types.Exporter {
+	if npdo.PrometheusServerPort <= 0 {
+		return nil
+	}
+
+	addr := net.JoinHostPort(npdo.PrometheusServerAddress, strconv.Itoa(npdo.PrometheusServerPort))
+	pe, err := prometheus.NewExporter(prometheus.Options{})
+	if err != nil {
+		glog.Fatalf("Failed to create Prometheus exporter: %v", err)
+	}
+	go func() {
+		mux := http.NewServeMux()
+		mux.Handle("/metrics", pe)
+		if err := http.ListenAndServe(addr, mux); err != nil {
+			glog.Fatalf("Failed to start Prometheus scrape endpoint: %v", err)
+		}
+	}()
+	view.RegisterExporter(pe)
+	return &prometheusExporter{}
+}
+
+// ExportProblems does nothing.
+// Prometheus exporter ignores all problems, and only deal with metrics.
```

Comment:
I'm not sure ignores all problems is a good idea, I think problem is a kind of metrics.
There is a case: The remedy system gets problem from prometheus or other monitoring system, then start remedy job, such as restart some processes, reboot node. 
So if NPD could export the problem as metric would be much helpful.

Reply:
Hi Juan,

Thanks for the review! And I totally agree that NPD should export all current problems as metrics.
The detailed design of how I plan to achieve that is at the "[Extending Today’s Problem Daemon to Report Metrics](https://docs.google.com/document/d/1SeaUz6kBavI283Dq8GBpoEUDrHA2a795xtw0OvjM568/edit#heading=h.eym0m8duezae)" section of the [proposal](https://docs.google.com/document/d/1SeaUz6kBavI283Dq8GBpoEUDrHA2a795xtw0OvjM568/edit).

The basic idea is, problem daemons (today we only have two, System Log Monitor and Custom Plugin Monitor) will convert problems to metrics (counter, or gauge). For example:
`problem_counter{"name": "DockerRestart"}  17`

The current [config files](https://github.com/kubernetes/node-problem-detector/tree/v0.6.3/config) won't need any modification to support this. Default metrics conversion will be provided at problem daemon level. Of course, if fancier conversion is needed (e.g. renaming the metric name/label), we could allow those to be configurable as well.

Since problem daemons will convert problems into metrics and report metrics directly, Prometheus exporter can just happy take the _metrics_, and ignore all _problems_ :)

The line `// Prometheus exporter ignores all problems, and only deal with metrics` actually means that Prometheus exporter do not report _problems_, but only report _metrics_. The notion of _problems_ and _metrics_ is newly introduced in the [proposal](https://docs.google.com/document/d/1SeaUz6kBavI283Dq8GBpoEUDrHA2a795xtw0OvjM568/edit). Please see the "[Detailed NPD architecture](https://docs.google.com/document/d/1SeaUz6kBavI283Dq8GBpoEUDrHA2a795xtw0OvjM568/edit#heading=h.jhsmauthxd88)" and "[Core Data Representation Model](https://docs.google.com/document/d/1SeaUz6kBavI283Dq8GBpoEUDrHA2a795xtw0OvjM568/edit#heading=h.tlefaa9eq3qb)" sections.

Reply:
Thank you for prompt reply. It make sense to me.

Reply:
Thanks Juan :)

Reply:
Then please change the comment. :)

Reply:
Hum...I feel the comment was correct...
Changed it to `// Prometheus exporter only exports metrics.`
Do you think this would reduce the confusion? Thanks!